resources/providers/awslib/iam/user.go

```diff
@@ -165,55 +162,118 @@ func (p Provider) getMFADevices(ctx context.Context, user types.User) ([]AuthDev
 	return devices, nil
 }
 
-func (p Provider) getUserKeys(ctx context.Context, apiUser types.User) ([]AccessKey, error) {
+func (p Provider) getUserKeys(username string, report map[string]*CredentialReport) []AccessKey {
+	p.log.Debugf("aggregate access keys data for user: %s", username)
+	entry := report[username]
 
-	var keys []AccessKey
-	input := iamsdk.ListAccessKeysInput{
-		UserName: apiUser.UserName,
+	if entry == nil {
+		p.log.Debugf("no entry for user: %s in credentials report", username)
+		return nil
 	}
-	for {
-		output, err := p.client.ListAccessKeys(ctx, &input)
-		if err != nil {
-			return nil, err
-		}
 
-		for _, apiAccessKey := range output.AccessKeyMetadata {
-			output, err := p.client.GetAccessKeyLastUsed(ctx, &iamsdk.GetAccessKeyLastUsedInput{
-				AccessKeyId: apiAccessKey.AccessKeyId,
-			})
+	return []AccessKey{
+		{
+			Active:       entry.AccessKey1Active,
+			LastAccess:   entry.AccessKey1LastUsed,
+			HasUsed:      entry.AccessKey1LastUsed != "N/A",
+			RotationDate: entry.AccessKey1LastRotated,
+		}, {
+			Active:       entry.AccessKey2Active,
+			LastAccess:   entry.AccessKey2LastUsed,
+			HasUsed:      entry.AccessKey2LastUsed != "N/A",
+			RotationDate: entry.AccessKey2LastRotated,
+		},
+	}
+}
 
-			var lastUsed time.Time
-			if err == nil {
-				if output.AccessKeyLastUsed != nil && output.AccessKeyLastUsed.LastUsedDate != nil {
-					lastUsed = *output.AccessKeyLastUsed.LastUsedDate
+func (p Provider) getCredentialReport(ctx context.Context) (map[string]*CredentialReport, error) {
+	var ae smithy.APIError
+	report, err := p.client.GetCredentialReport(ctx, &iamsdk.GetCredentialReportInput{})
+	if err != nil {
+		var awsFailErr *types.ServiceFailureException
+		if errors.As(err, &awsFailErr) {
+			return nil, errors.Wrap(err, "could not gather aws iam credential report")
+		}
+
+		// if we have an error, and it is not a server err we generate a report
+		if errors.As(err, &ae) {
+			if ae.ErrorCode() == "ReportNotPresent" || ae.ErrorCode() == "ReportExpired" {
```

Comment:
`ReportNotPresent` and `ReportExpired` are not exported as consts?

Reply:
Unfortunately, the ErrCodeNoSuchEntityException error code does not exist in the v2 SDK.